skywalking/plugins/__init__.py

```diff
@@ -38,7 +41,65 @@ def install():
             continue
         logger.debug('installing plugin %s', modname)
         plugin = importer.find_module(modname).load_module(modname)
+
+        logger.debug('checking version for plugin %s', modname)
+        supported = pkg_version_check(plugin, modname)
+        if not supported:
+            continue
+
         if not hasattr(plugin, 'install') or inspect.ismethod(getattr(plugin, 'install')):
             logger.warning('no `install` method in plugin %s, thus the plugin won\'t be installed', modname)
             continue
         plugin.install()
+
+
+_operators = {
+    '<': lambda cv, ev: cv < ev,
+    '<=': lambda cv, ev: cv < ev or cv == ev,
+    '=': lambda cv, ev: cv == ev,
+    '>=': lambda cv, ev: cv > ev or cv == ev,
+    '>': lambda cv, ev: cv > ev,
+    '!=': lambda cv, ev: cv != ev
+}
+
+
+def pkg_version_check(plugin, modname):
+    supported = True
+
+    # no version rules was set, no checks
+    if not hasattr(plugin, "version_rule"):
+        return supported
+
+    pkg_name = plugin.version_rule.get("name")
+    rules = plugin.version_rule.get("rules")
+
+    try:
+        current_pkg_version = pkg_resources.get_distribution(pkg_name).version
+    except pkg_resources.DistributionNotFound:
+        supported = False
+        logger.warning("plugin %s didn\'t find the corresponding package %s, thus won't be installed",
+                       modname, pkg_name)
```

Comment:
When failed to get the version, we can just consider it as supported as the built-in libraries should always throw exception here